src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Search/Behaviors/ListSearchBehaviorTests.cs

```diff
@@ -0,0 +1,216 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Serialization;
+using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Context;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Features.Search.Behavior;
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.Core.Messages.Search;
+using Microsoft.Health.Fhir.Core.Models;
+using Microsoft.Health.Fhir.Tests.Common;
+using NSubstitute;
+using Xunit;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Core.UnitTests.Features.Search
+{
+    public class ListSearchBehaviorTests
+    {
+        private readonly ResourceDeserializer _resourceDeserializer;
+
+        private readonly FhirJsonParser _fhirJsonParser = new FhirJsonParser();
+
+        private readonly IFhirDataStore _fhirDataStore;
+
+        private IScoped<IFhirDataStore> _scopedDataStore;
+
+        private readonly CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
+
+        private readonly CancellationToken _cancellationToken;
+
+        private readonly ISearchOptionsFactory _searchOptionsFactory;
+
+        private readonly IBundleFactory _bundleFactory;
+
+        private ResourceElement _nonEmptyBundle;
+
+        public ListSearchBehaviorTests()
+        {
+            _cancellationToken = _cancellationTokenSource.Token;
+
+            var so = new SearchOptions();
+            so.UnsupportedSearchParams = new Tuple<string, string>[0];
+            so.UnsupportedSortingParams = Array.Empty<(string searchParameterName, string reason)>();
+
+            _searchOptionsFactory = Substitute.For<ISearchOptionsFactory>();
+            _searchOptionsFactory.Create(Arg.Any<string>(), Arg.Any<IReadOnlyList<Tuple<string, string>>>()).Returns(so);
+
+            _resourceDeserializer = new ResourceDeserializer(
+               (FhirResourceFormat.Json, new Func<string, string, DateTimeOffset, ResourceElement>((str, version, lastUpdated) => _fhirJsonParser.Parse(str).ToResourceElement())));
```

Comment:
There is a test helper Microsoft.Health.Fhir.Tests.Common.ResourceDeserializer that you should be able to use here